enhancements/sandboxed-containers/sandboxed-containers-tech-preview.md

```diff
@@ -0,0 +1,620 @@
+
+Sandboxed containers relies, under the hood, on [Kata
+Containers](https://katacontainers.io/), an Open Source project able to
+achieve all the proposed points.
+
+It should be noted that Kata Containers differs from KubeVirt:
+
+- KubeVirt aims to run **VM images** on OpenShift providing the VM with the
+  look and feel of a virtual machine running as a legacy VM (CRD defining a
+  Virtual Machine in Kubernetes).
+- Kata Containers aims to run **container images** on OpenShift in an
+  isolated manner using virtualization tools (uses Runtime Class resource to
+  choose Kata Containers only for specific workloads which require kernel
+  level isolation).
+  
+## Motivation
+
+The main motivation to bring sandboxed containers to OpenShift is to provide
+an extra layer of isolation for fields that would commonly require that, such
+as Telco companies.
+
+### Goals
+
+- Provide a way to enable running user workloads on sandboxed containers on an
+  OpenShift cluster;
+- Clusters with Sandboxed Container workloads should support upgrades;
+- Add security policy to control which users can run what workloads on the
+  sandboxed containers in an OpenShift cluster.
+
+### Non Goals
+
+- Running workloads using a different kernel than the one used on the
+  OpenShift node
+  + Although sandboxed containers are capable of supporting running a
+    different kernel then the one used on the OpenShift node, such features
+    will not be supported in order to simplify the development cycles and
+    testing efforts of this project;
+- Running OpenShift control plane and core operators on Sandboxed Containers
+  + These will continue to run on runc.
+
+## Proposal
+
+As part of our proposal, we'll cover a few user stories coming from the Telco
+industry.
+
+### User Stories
+
+#### Story 1 - Telco CNF deployments
+
+As part of 5G/NFV telco deployments there is a gradual migration from
+physical networking boxes to CNFs (container network functions). Some of
+these CNFs require root access while others could potentially create a threat
+on the operators cloud bypassing existing container isolation mechanisms.
+For addressing those issues a hardened isolated container solution is
+required.
+
+#### Story 2 - Devops deployments
+
+Telcos are working hard to increase the rate of adding new capabilities to
+their clouds for competing with OTTs (over the top companies such as Netflix,
+Apple, Facebook etc…). This involves huge investments in devops tools and
+processes. In order for devops to deploy new features on production
+environments there is a need for high isolation of the deployed workloads in
+order to control and revert such changes.
+
+#### Story 3 - Hardware Vendors
+
+Many hardware vendors require custom kernel parameters at boot, custom sysctl
+variables, etc. They want to use containers as a convenient packaging format
+for their entire application, but still need access to lower level tuning for
+portions of their applications.
```

Comment:
nit: Could add a use case for builds that we don't have to tackle right away but is enabled by this work.